src/test/java/rx/internal/operators/OperatorObserveOnTest.java

```diff
@@ -804,5 +783,55 @@ public void onNext(Integer t) {
         assertTrue(latch.await(10, TimeUnit.SECONDS));
         assertEquals(1, requests.size());
     }
+
+    @Test
+    public void testErrorDelayed() {
+        TestScheduler s = Schedulers.test();
+        
+        Observable<Integer> source = Observable.just(1, 2 ,3)
+                .concatWith(Observable.<Integer>error(new TestException()));
+        
+        TestSubscriber<Integer> ts = TestSubscriber.create(0);
+
+        source.observeOn(s, true).subscribe(ts);
+        
+        ts.assertNoValues();
+        ts.assertNoErrors();
+        ts.assertNotCompleted();
+        
+        s.advanceTimeBy(1, TimeUnit.SECONDS);
+
+        ts.assertNoValues();
+        ts.assertNoErrors();
+        ts.assertNotCompleted();
+
+        ts.requestMore(1);
+        s.advanceTimeBy(1, TimeUnit.SECONDS);
+        
+        ts.assertValues(1);
+        ts.assertNoErrors();
+        ts.assertNotCompleted();
+        
+        ts.requestMore(3); // requesting 2 doesn't switch to the error() source for some reason in concat.
```

Comment:
The drain loop quits because it runs out of requests but at the same time, the concatenated data source is also empty. I can add a `if (checkTerminated(finished, q.isEmpty(), localChild, q)) return` check again, but this adds overhead for those sources that request again anyway.